### PR TITLE
EHN: adding subtract paddle.tenor.tensor.Tensor

### PR DESCRIPTION
**Pull request by MuhammadNizamani**
Adding subtrack function paddle.tenor.tensor.Tensor from issue  #16452

### CODE REVIEW

**Review by khushi-411 — CHANGES_REQUESTED** [review]
Hi, @MuhammadNizamani! Thanks for the PR. I see there are a couple of merge conflicts and lint failures. Could you please resolve those failures? Thanks!

**Comment by MuhammadNizamani** [comment]
@khushi-411  would you please help to understand test case in this PR.

File: ivy/functional/frontends/paddle/tensor/tensor.py [hunk 1]
```diff
@@ -139,6 +139,10 @@ def exp(self, name=None):
         return ivy.exp(self._ivy_array)
 
     @with_unsupported_dtypes({"2.4.2 and below": ("float16", "bfloat16")}, "paddle")
+    def subtract(self, other, name=None):
+        other_ivy = _to_ivy_array(other)
+        return ivy.subtract(self._ivy_array, other_ivy)
+
```
Comment (khushi-411) [inline ivy/functional/frontends/paddle/tensor/tensor.py:145]: You should add a new line with the decorator instead of using the previous one.

Comment (MuhammadNizamani) [reply ivy/functional/frontends/paddle/tensor/tensor.py:145]: @khushi-411  I don't understand the what you want me to do. Could you please elaborate?

**Comment by MuhammadNizamani** [comment]
@khushi-411 check this PR

File: ivy/functional/frontends/paddle/tensor/tensor.py (hunk 1 above)
Comment (khushi-411) [reply ivy/functional/frontends/paddle/tensor/tensor.py:147]: Please carefully check and address these comments. Notice the 6th line and check your functions.

**Comment by khushi-411** [comment]
There are many failures. Have you been able to check if the failures are related or not? If not, please rebase and ping me. I'll merge the PR. Thanks!

**Comment by MuhammadNizamani** [comment]
Failures are from my side the issue is that the test function is not working properly.
And can you review my  PR other 2 because mehmod is assigned to the PR but he says that he is not assigned to review frontend PR.

**Comment by MuhammadNizamani** [comment]
Sure , 
https://github.com/unifyai/ivy/pull/16991
This is the PR.

**Comment by MuhammadNizamani** [comment]
@khushi-411  I have made some changes in test function and the subtract function now this PR passing all local test.

File: ivy/functional/frontends/paddle/tensor/tensor.py (hunk 1 above)
Comment (khushi-411) [reply ivy/functional/frontends/paddle/tensor/tensor.py]: This point still needs to be addressed

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py [hunk 2]
```diff
@@ -1026,3 +1026,34 @@ def test_paddle_erf(
...
     )
+
+
+# subtract
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="paddle.to_tensor",
+    method_name="subtract",
+    dtypes_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"), num_arrays=2, shared_dtype=True
```
[the remark this answers is not in the document]
Comment (MuhammadNizamani) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:1037]: Checkout the video that you sent me and I he suggested that so I added it.

[the remark this answers is not in the document]
Comment (MuhammadNizamani) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:1037]: @khushi-411  should I remove it ?

[the remark this answers is not in the document]
Comment (khushi-411) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:1037]: I think it's fine. Let's keep it.

**Review by khushi-411 — APPROVED** [review]
LGTM! Thanks, @MuhammadNizamani, for the PR!

**Review by khushi-411 — CHANGES_REQUESTED** [review]
Whoops, unfortunately, there are merge conflicts. Could you resolve it?

**Comment by MuhammadNizamani** [comment]
Sure

**Comment by MuhammadNizamani** [comment]
@khushi-411 done

**Review by khushi-411 — APPROVED** [review]
LGTM, Thanks! The failures are unrelated.

**Pull request merged**